extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/StartupBuildSteps.java

```diff
@@ -62,7 +64,7 @@ public boolean appliesTo(org.jboss.jandex.AnnotationTarget.Kind kind) {
 
             @Override
             public void transform(TransformationContext context) {
-                if (context.isClass() && !BuiltinScope.isDeclaredOn(context.getTarget().asClass())) {
+                if (context.isClass() && !isDeclaredOn(context.get(Key.SCOPES_NAMES), context.getTarget().asClass())) {
```

Comment:
I can see a little inconsistency here - build steps can simply inject the list of `ContextRegistrarBuildItem`, ie. `Key.SCOPES_NAMES` is not needed. However, regular ArC extensions have no way to register custom annotations and so `Key.SCOPES_NAMES` is not very useful outside quarkus.

Reply:
Note that `KEY.SCOPE_NAMES` is all scopes that ArC knows, not just custom one. That allows any Quarkus-ArC extension to check the list of all known scopes at (almost) any point and using the same way that you would use to grab annotation store, injection points, beans.... Without it, you need to work with a combination of built in scopes and context registrar items. What I attempted here just seemed a lot more user friendly to me.